Evo recs

kingomes opened the pull request:
- Switched to button-based giveaway
- Slash command has autocomplete for time options (the modal does not though)
- Giveaway can be edited
- Giveaway can be ended early

kingomes committed:
started to add fail buttons but new issues to resolve

kingomes committed:
Merge remote-tracking branch 'origin/main' into giveaway-retrybuttons

kingomes committed:
got rid of resend button

kingomes committed:
retry buttons improved

kingomes committed:
giveaway reroll error handling

kingomes committed:
proper use of stmt.filter

kingomes committed:
fixed reroll fail message

kingomes committed:
using prettier to format

kingomes committed:
got rid of giveaway reroll

kingomes committed:
switched to button-based giveaway

kingomes committed:
number of entries stays updated

kingomes committed:
working towards edit + end buttons (broken for more than one giveaway at a time)

kingomes committed:
enter and leave giveaway should work with multiple giveaways (not end giveaway)

kingomes committed:
giveawayEnd should work + database columns adjusted

kingomes committed:
giveawayEdit should work (redundant code though)

kingomes committed:
autocomplete functional for giveaway command

kingomes committed:
re-added owner restrictions

kingomes committed:
small autocomplete variables fix

EvolutionX-10 commented on src/commands/handlers/giveawayDiscard.ts:
```diff
@@ -0,0 +1,21 @@
+import { ownerIDs } from "#constants";
+import { commandModule, CommandType } from "@sern/handler";
+import { db } from "../../utils/db.js";
```
path alias?

EvolutionX-10 commented on src/commands/handlers/giveawayEditModal.ts:
```diff
@@ -0,0 +1,95 @@
+import { commandModule, CommandType } from "@sern/handler";
+import { ownerIDs } from "#constants";
+import { db } from "../../utils/db.js";
```
path alias?

EvolutionX-10 commented on src/commands/handlers/giveawayEnd.ts:
```diff
@@ -0,0 +1,87 @@
+import { commandModule, CommandType } from "@sern/handler";
+import { db } from "../../utils/db.js";
```
path alias?

EvolutionX-10 commented on src/commands/handlers/giveawayEnd.ts:
```diff
@@ -0,0 +1,87 @@
+import { commandModule, CommandType } from "@sern/handler";
+import { db } from "../../utils/db.js";
+import { ownerIDs } from "#constants";
+import { discardRows } from "../giveaway.js";
```
path alias?

EvolutionX-10 commented on src/commands/handlers/giveawayEnter.ts:
```diff
@@ -0,0 +1,66 @@
+import { commandModule, CommandType } from "@sern/handler";
+import { db } from "../../utils/db.js";
```
path alias?

EvolutionX-10 commented on src/commands/handlers/giveawayLeave.ts:
```diff
@@ -0,0 +1,61 @@
+import { commandModule, CommandType } from "@sern/handler";
+import { db } from "../../utils/db.js";
```
path alias?

EvolutionX-10 reviewed (commented):
Assuming it's tested properly, looks good to me
Please do the mentioned changes in imports, and we'll be good to go!

kingomes committed:
using aliases + formatted with prettier

kingomes committed:
corrected aliases

kingomes commented:
i just added the aliases and made sure the prettier formatting is back in tact. i added a #db alias, which goes straight to the file and a #commands alias, which goes to the commands folder.

i also did keep testing the additions as i added them, so i hope i caught most of the possible issues that could occur with them.

one potential issue is that the giveaway command has autocomplete but the giveaway edit modal, according to the internet, does not support autocomplete, so if someone wants to edit the time, they will have to type it out the same way the command was before the autocomplete was added (i just added an extra check to at least make sure that if the one editing the giveaway mistypes it, it should just ask them to try again with a valid time instead of immediately ending the giveaway).

kingomes committed:
added extra check to edit modal
